{% extends 'base.html' %}

{% load crispy_forms_tags %}

{% block content %}
    <div class="col-md-12">
        <h3 class="mb-3">Home</h3>
        <hr>
        <div class="mb-3" style="text-align: justify;"><p>EpiGe-app is a new tool for the analysis of the methylation status of unique CpG sites based on real-time PCR allelic discrimination technology. Taking advantage of the specific, bimodal methylation values of the six-cytosine biomarkers that compose our <sup>Epi</sup>WNT-SHH classifier <a href="https://pubmed.ncbi.nlm.nih.gov/29351917/" target="_blank" rel="noopener noreferrer">[GÃ³mez S et al. Clin Cancer Res. 2018 Mar 15;24(6):1355-1363. doi: 10.1158/1078-0432.CCR-17-2243]</a>, we have developed subgroup-specific fluorogenic probes that permit to discriminate methylated and unmethylated single cytosine residues. We called this technical adaptation <i>EpiGe</i>. For the interpretation of the EpiGe qPCR results, we have designed this interactive Web Application named EpiGe-app. Once the bisulfite allele-specific probes are detected using real-time polymerase chain reaction (PCR)-based assays, the EpiGe-app enables the automatic analysis of genotype calls and assigns each sample to a category (subgroup).</p></div>
        <div class="alert alert-danger" role="alert">
            Classification using methylation profiling is a research tool under development, it is not verified and has not been clinically validated. Implementation of the results in a clinical setting is in the sole responsibility of the treating physician.
        </div>
        <a class="btn btn-primary btn-lg btn-block" href="{% url 'core_app:protocols' %}" role="button">Start</a>
    </div>
{% endblock %}