{% extends 'base.html' %}
{% load static %}

{% load crispy_forms_tags %}

{% block content %}
    <h3>EpiGe Medulloblastoma Subgroup Prediction (v1.0)</h3>
    <hr>
    <div class="card" style="width: 30rem;">
        <div class="card-header"><b>Basic Information</b></div>
        <ul class="list-group list-group-flush">
            <li class="list-group-item"><b>Sample ID:</b> {{ sample.sample_identifier }}</li>
            <li class="list-group-item"><b>Uploaded file:</b> {{ sample.filename}}</li>
            <li class="list-group-item"><b>Size file:</b> {{ sample.filesize}}</li>
            {% if sample.send_email %}
              <li class="list-group-item"><b>Email:</b> {{ sample.email }}</li>
            {% endif %}
            <li class="list-group-item"><b>Date:</b> {{ sample.created|date:"M d, Y, H:i:s" }}</li>
        </ul>
    </div>
    <br>

    {% if sample.status == 1 %}

      <h3>Medulloblastoma Subgroup Assignment</h3>
      <hr>
      {% if classification.subgroup == "WNT" %}
            <h4 style="margin-bottom: 0px;">Medulloblastoma, Wingless/INT1 (WNT)-activated</h4>
            <p style="text-align: justify;text-justify: inter-word;margin-top: 0px;">
                WNT-activated medulloblastoma is an embryonal tumour thought to arise from cells in the dorsal brainstem. 
                WNT-activated tumours account for approximately 10% of all medulloblastomas, occurring generally in children between 4 and 7 years of age, and accounting for 15-20% of adult medulloblastomas. 
                Two molecular subgroups have been described: WNT-α and WNT-β that differ in age at presentation of the disease, and frequency of monosomy 6. 
                Patients with WNT-activated medulloblastomas rarely present metastatic disease at diagnosis, and the outcomes are favourable (5-year survival of more than 95%). 
                These tumours generally have a classic histology. Somatic mutations in exon 3 of CTNNB1 gene are present in 85-89% of WNT-activated medulloblastomas. 
                Most WNT-activated medulloblastoma lacking somatic CTNNB1 mutation arise in patients carrying pathogenic germline <i>APC</i> mutations. 
                Monosomy of chromosome 6, in absence of other somatic copy-number alterations, is a characteristic genetic feature of these tumours.
            </p>
      {% elif classification.subgroup == "SHH" %}
            <h4 style="margin-bottom: 0px;">Medulloblastoma, Sonic hedgehog (SHH)-activated</h4>
            <p style="text-align: justify;text-justify: inter-word;margin-top: 0px;">
                SHH-activated medulloblastoma is an embryonal tumour of the cerebellum, characterized by the activation of the SHH signalling pathway. 
                SHH-activated tumours represent about 30% of patients with medulloblastoma, and are subdivided according to <i>TP53</i> gene mutation status (SHH-with wild type <i>TP53</i> and SHH with mutant <i>TP53</i>). 
                SHH-activated medulloblastoma display a bimodal age distribution, being the most common molecular subgroup in infants (less than 3 years of age) and adults (older than 17 years), with fewer cases being diagnosed during childhood and adolescence. 
                Medulloblastoma SHH are characterized by upstream SHH pathway alterations, such as PTCH1 and SMO, or SUFU. Young MB SHH patients may have germline origin associated to Gorlin’s syndrome. Histologically most cases present as desmoplastic or extensive nodularity variant, less commonly as classic variant and occasionally as large cell/anaplastic. 
                SHH-activated tumours are located in the cerebellum with a tendency to be located laterally. 
                Median age of presentation is 2 years for infants and 22 years for children and adult. 
                The prognosis of SHH-activated MBs is largely dependent on patient’s age and specific genetic features where children with TP53 mutated SHH tumours have poorer outcome.
            </p>
      {% elif classification.subgroup == "non-WNT/non-SHH" %}
            <h4 style="margin-bottom: 0px;">Medulloblastoma, non-WNT/non-SHH</h4>
            <p style="text-align: justify;text-justify: inter-word;margin-top: 0px;">
                Medulloblastoma, non-WNT/non-SHH is an embryonal tumour of the cerebellum without a molecular signature associated with activation of the WNT or SHH signalling pathway.
                Non-WNT/non-SHH are classified as group 3 and group 4 tumours and comprise eight molecular subgroups. 
                Group 3 tumours occur during infancy and childhood, and account for about 25% of medulloblastomas (~40% of cases in infants). 
                Group 4 medulloblastomas occurs across all age groups, accounting for 40% of cases. 
                Approximately 30-40% of patients of both subgroups, present metastasis disease at the moment of diagnosis. 
                Histologically most cases fall into the classical and large cell/anaplastic groups. 
                Group 3 and group 4 are characterized by partly overlapping genetic features, making their distinction in some cases difficult. 
                Consequently, group 3 and group 4 have been described as a single subgroup designated non-WNT/non-SHH. 
                <i>MYC</i> or <i>MYCN</i> amplification, isochromosome 17q (found in >50%) and GFI1/1B activation by enhancer hijacking are recurrent features in both subgroups. 
                <i>MYC</i> amplification is a common genetic alteration seen in group 3 (17%) medulloblastomas, and is associated with a poor outcome. 
                <i>MYCN</i> amplification in the most frequent alteration identified in group 4 medulloblastomas. 
                Metastatic disease at presentation of the disease, is associated with poor outcome in group 4 medulloblastomas.
            </p>
      {% else %}
          <p class="text-align: justify;text-justify: inter-word;"></p>
      {% endif %}
      <h4>Disclaimer</h4>
      <div class="alert alert-danger" role="alert">
        <p class="pb-0 mb-0" style="text-align: justify;"><i class="fa-solid fa-triangle-exclamation mr-1"></i> EpiGe-App is a research tool that is intended only for scientific purposes: the EpiGe-App has not been validated clinically and was not designed for diagnostics purpose, nor to replace the services of a licensed, trained physician or health professional, or to substitute for medical advice.</p>
      </div>
      <br>
      <h3>Cytosine Methylation Status Predictor</h3>
      <hr>
      <h5>Table of reference patterns</h5>
      <figure class="figure border bg-white">
        <img src="{% static 'img/reference_table.png' %}" class="figure-img img-fluid rounded" alt="reference_table_image">
      </figure>
      <h5>Table of probabilities of being methylated obtained by logistic regression</h5>
      <figure class="figure border bg-white">
        <img src="{% get_static_prefix %}samples/{{ sample.id }}/probability_table.png" class="figure-img img-fluid rounded" alt="probabilities_table_image">
      </figure>
      <h5>Table of sample pattern obtained by logistic regression</h5>
      <figure class="figure border bg-white">
        <img src="{% get_static_prefix %}samples/{{ sample.id }}/sample_table.png"  class="figure-img img-fluid rounded" alt="sample_table_image">
      </figure>
      <br>
      <h3>Medulloblastoma Subgroup Classification by Hamming Distance</h3>
      <hr>
      <h5>Radar plot with Hamming distances</h5>
      <figure class="figure">
          <img src="{% get_static_prefix %}samples/{{ sample.id }}/radar_plot.png" class="figure-img img-fluid rounded mx-auto d-block border" alt="radar_plot" style="width:50%">
      </figure>
      <h5>Table of scores</h5>
      <table class="table border bg-white">
          <thead>
            <tr>
              <th scope="col" class="text-center">Molecular subgroup</th>
              <th scope="col" class="text-center">Score for WNT</th>
              <th scope="col" class="text-center">Score for SHH</th>
              <th scope="col" class="text-center">Score for non-WNT/non-SHH</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row" class="text-center">{{ classification.subgroup }}</th>
              <td class="text-center">{{ classification.score_wnt }}</td>
              <td class="text-center">{{ classification.score_shh }}</td>
              <td class="text-center">{{ classification.score_gg }}</td>
            </tr>
          </tbody>
      </table>
      <br>
      <h3>Quality Control Summary</h3>
      <hr>
      <table class="table border bg-white">
          <thead>
            <tr>
              <th scope="col">CheckPoint</th>
              <th scope="col">Result</th>
              <th scope="col">Possible Solution</th>
            </tr>
          </thead>
          <tbody>
              <tr>
                  <th scope="row">Text file complete</th>
                  <td>The txt file contains all the information.</td>
                  <td></td>
              </tr>
              <tr>
                  <th scope="row">All CpG</th>
                  <td>All 6 CpG are included.</td>
                  <td></td>
              </tr>
              <tr>
                  <th scope="row">Dye calibration</th>
                  {% if calibration.ROX_valid and calibration.VIC_valid and calibration.FAM_valid %}
                      <td>All dyes are calibrated.</td>
                      <td></td>
                  {% else %}
                      <td>Some of the dyes have expired calibration.</td>
                      <td>See the technical report for more information.</td>
                  {% endif %}
              </tr>
              <tr>
                  <th scope="row">No template controls</th>
                  {% if not calibration.amplification_test %}
                      <td>Amplification has been detected in some no template controls.</td>
                      <td>See the technical report for more information.</td>
                  {% else %}
                      <td>No amplification was detected in any no template control.</td>
                      <td></td>
                  {% endif %}
              </tr>
              <tr>
                  <th scope="row">SD<0.5</th>
                  {% if calibration.standard_deviation_test %}
                      <td>Some pairs of replicates have a standard deviation greater than 0.5.</td>
                      <td>See the technical report for more information.</td>
                  {% else %}
                      <td>No pairs of replicates have a standard deviation greater than 0.5.</td>
                      <td></td>
                  {% endif %}
              </tr>
          </tbody>
      </table>
      <br>
      <h3>Technical report</h3>
      <hr>
      <h4>Calibration information</h4>
      <table class="table border bg-white">
          <thead>
          <tr>
              <th>ROX is expired</th>
              <th>ROX performed</th>
              <th>FAM is expired</th>
              <th>FAM performed</th>
              <th>VIC is expired</th>
              <th>VIC performed</th>
          </tr>
          </thead>
          <tbody>
              <tr>
                  <td>{% if calibration.ROX_valid %}No{% else %}Yes{% endif %}</td>
                  <td>{{ calibration.ROX_date }}</td>
                  <td>{% if calibration.FAM_valid %}No{% else %}Yes{% endif %}</td>
                  <td>{{ calibration.FAM_date }}</td>
                  <td>{% if calibration.VIC_valid %}No{% else %}Yes{% endif %}</td>
                  <td>{{ calibration.VIC_date }}</td>
              </tr>
          </tbody>
      </table>
      <h4>No template control amplification</h4>
      <figure class="figure border bg-white">
        <img src="{% get_static_prefix %}samples/{{ sample.id }}/NTC_table.png"  class="figure-img img-fluid rounded" alt="ntc_table_image">
      </figure>
      {% if calibration.amplification_test %}
          <p>No amplification detected.</p>
      {% else %}
          <p>Amplification detected.</p>
      {% endif %}
      <h4>SD<0.5</h4>
      <figure class="figure border bg-white">
        <img src="{% get_static_prefix %}samples/{{ sample.id }}/standard_deviation_table.png" class="figure-img img-fluid rounded" alt="standard_deviation_image">
      </figure>
      {% if calibration.standard_deviation_test %}
          <p>Some pairs of replicates have a standard deviation greater than 0.5.</p>
      {% else %}
          <p>No pairs of replicates have a standard deviation greater than 0.5.</p>
      {% endif %}
      <br>
      <h3>Run information</h3>
      <hr>
      <table class="table border bg-white">
          <thead>
            <tr>
              <th scope="col">#</th>
              <th scope="col">Task</th>
              <th scope="col">Version</th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <th scope="row">1</th>
              <td>EpiGe_qc</td>
              <td>1.0</td>
            </tr>
            <tr>
              <th scope="row">2</th>
              <td>EpiGe_logisticRegression</td>
              <td>1.0</td>
            </tr>
            <tr>
              <th scope="row">3</th>
              <td>EpiGe_hammingDistance</td>
              <td>1.0</td>
            </tr>
            <tr>
              <th scope="row">4</th>
              <td>EpiGe_report</td>
              <td>1.0</td>
            </tr>
          </tbody>
      </table>
      <br>
      <h3>References</h3>
      <hr>
      <ol>
          <li><a id="cita_louis" href="https://pubmed.ncbi.nlm.nih.gov/34185076/" target="_blank" rel="noopener noreferrer">Louis DN, Perry A, Wesseling P, Brat DJ, Cree IA, Figarella-Branger D, Hawkins C, Ng HK, Pfister SM, Reifenberger G, Soffietti R, von Deimling A, Ellison DW. The 2021 WHO Classification of Tumors of the Central Nervous System: a summary. Neuro Oncol. 2021 Aug 2;23(8):1231-1251. doi: 10.1093/neuonc/noab106. PMID: 34185076; PMCID: PMC8328013.</a></li><br>
          <li><a href="https://publications.iarc.fr/Book-And-Report-Series/Who-Classification-Of-Tumours/Central-Nervous-System-Tumours-2021" target="_blank" rel="noopener noreferrer" style="text-align: justify;text-justify: inter-word;">WHO Classification of Tumours. Central Nervous System Tumours/Fifth Edition, IARC 2021.</a></li><br>
          <li><a href="https://pubmed.ncbi.nlm.nih.gov/29351917/" target="_blank" rel="noopener noreferrer">Gómez S, Garrido-Garcia A, Garcia-Gerique L, Lemos I, Suñol M, de Torres C, Kulis M, Pérez-Jaume S, Carcaboso ÁM, Luu B, Kieran MW, Jabado N, Kozlenkov A, Dracheva S, Ramaswamy V, Hovestadt V, Johann P, Jones DTW, Pfister SM, Morales La Madrid A, Cruz O, Taylor MD, Martin-Subero JI, Mora J, Lavarino C. A Novel Method for Rapid Molecular Subgrouping of Medulloblastoma. Clin Cancer Res. 2018 Mar 15;24(6):1355-1363. doi: 10.1158/1078-0432.CCR-17-2243. Epub 2018 Jan 19. PMID: 29351917.</a></li>
      </ol>
      <br>
      <h3>Download</h3>
      <hr>
      <div class="container">
        <div class="col-md-12 text-center">
            <a class="btn btn-success btn-lg" href="{% url 'workflow:download_report' sample.id %}" role="button" target="_blank" download>Download as PDF file <i class="fa-solid fa-file-arrow-down ml-2"></i></a>
        </div>
      </div>
    
    {% elif sample.status == 2 %}

      <h3>Results</h3>
      <hr>
      <p>The uploaded file is not complete.</p>
      <br>
      <h3>Quality Control Summary</h3>
      <hr>
      <table class="table border bg-white">
          <thead>
          <tr>
              <th scope="col">CheckPoint</th>
              <th scope="col">Result</th>
              <th scope="col">Possible Solution</th>
          </tr>
          </thead>
          <tbody>
              <tr>
                  <td scope="row">TXT complete</td>
                  <td>The uploaded file is not complete.</td>
                  <td>Make sure that the file contains the following sections:
                      <ul>
                          <li>Amplification Data</li>
                          <li>Multicomponent Data</li>
                          <li>Raw Data</li>
                          <li>Reagent Information</li>
                          <li>Results</li>
                          <li>Sample Setup</li>
                      </ul>
                  </td>
              </tr>
              <tr>
                  <td scope="row">All CpG</td>
                  <td colspan="2">Not available.</td>
              </tr>
              <tr>
                  <td scope="row">Dye calibration</td>
                  <td colspan="2">Not available.</td>
              </tr>
              <tr>
                  <td scope="row">No template controls</td>
                  <td colspan="2">Not available.</td>
              </tr>
              <tr>
                  <td scope="row">SD<0.5</td>
                  <td colspan="2">Not available.</td>
              </tr>
          </tbody>
      </table>
      <br>
      <h3>Technical report</h3>
      <hr>
      <h4>Dye calibration</h4>
      <p>Not available.</p>
      <h4>No template control amplification</h4>
      <p>Not available.</p>
      <h4>SD<0.5</h4>
      <p>Not available.</p>
      <br>
      <h3>Run information</h3>
      <hr>
      <table class="table border bg-white">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">Task</th>
            <th scope="col">Version</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th scope="row">1</th>
            <td>EpiGe_qc</td>
            <td>1.0</td>
          </tr>
          <tr>
            <th scope="row">2</th>
            <td>EpiGe_logisticRegression</td>
            <td>1.0</td>
          </tr>
          <tr>
            <th scope="row">3</th>
            <td>EpiGe_hammingDistance</td>
            <td>1.0</td>
          </tr>
          <tr>
            <th scope="row">4</th>
            <td>EpiGe_report</td>
            <td>1.0</td>
          </tr>
        </tbody>
      </table>

    {% elif sample.status == 3 %}

      <h3>Results</h3>
      <hr>
      <p>Some CpGs are missing.</p>
      <p>Make sure that all six cytosines are in the file.</p>
      <p>Please check the cytosine indicated in the table below.</p>
      <br>
      <h3>Quality Control Summary</h3>
      <hr>
      <table class="table border bg-white">
          <thead>
          <tr>
              <th scope="col">CheckPoint</th>
              <th scope="col">Result</th>
              <th scope="col">Possible Solution</th>
          </tr>
          </thead>
          <tbody>
              <tr>
                  <td>TXT complete</td>
                  <td colspan="2">The txt file contains all the information.</td>
              </tr>
              <tr>
                  <td>All CpG</td>
                  <td>Some CpGs are missing.</td>
                  <td>We have not found the following cytosine: <b>{{ sample.missing_cpg }}</b></td>
              </tr>
              <tr>
                  <td>Dye calibration</td>
                  {% if calibration.ROX_valid and calibration.VIC_valid and calibration.FAM_valid %}
                      <td colspan="2">All dyes are calibrated.</td>
                  {% else %}
                      <td>Some of the dyes have expired calibration.</td>
                      <td>See the technical report for more information.</td>
                  {% endif %}
              </tr>
              <tr>
                  <td>No template controls</td>
                  {% if not calibration.amplification_test %}
                      <td>Amplification has been detected in some no template controls.</td>
                      <td>See the technical report for more information.</td>
                  {% else %}
                      <td colspan="2">No amplification was detected in any no template control.</td>
                  {% endif %}
              </tr>
              <tr>
                  <td>SD<0.5</td>
                  <td colspan="2">Not available.</td>
              </tr>
          </tbody>
      </table>
      <br>
      <h3>Technical report</h3>
      <hr>
      <h4>Dye calibration</h4>
      <table class="table border bg-white">
          <thead>
          <tr>
              <th>ROX is expired</th>
              <th>ROX performed</th>
              <th>FAM is expired</th>
              <th>FAM performed</th>
              <th>VIC is expired</th>
              <th>VIC performed</th>
          </tr>
          </thead>
          <tbody>
              <tr>
                  <td>{% if calibration.ROX_valid %}No{% else %}Yes{% endif %}</td>
                  <td>{{ calibration.ROX_date }}</td>
                  <td>{% if calibration.FAM_valid %}No{% else %}Yes{% endif %}</td>
                  <td>{{ calibration.FAM_date }}</td>
                  <td>{% if calibration.VIC_valid %}No{% else %}Yes{% endif %}</td>
                  <td>{{ calibration.VIC_date }}</td>
              </tr>
          </tbody>
      </table>
      <h4>No template control amplification</h4>
      <figure class="figure border bg-white">
        <img src="{% get_static_prefix %}samples/{{ sample.id }}/NTC_table.png"  class="figure-img img-fluid rounded" alt="ntc_table_image">
      </figure>
      {% if calibration.amplification_test %}
          <p>No amplification detected.</p>
      {% else %}
          <p>Amplification detected.</p>
      {% endif %}
      <h4>SD<0.5</h4>
      <p>Not available.</p>
      <br>
      <h3>Run information</h3>
      <hr>
      <table class="table border bg-white">
        <thead>
          <tr>
            <th scope="col">#</th>
            <th scope="col">Task</th>
            <th scope="col">Version</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <th scope="row">1</th>
            <td>EpiGe_qc</td>
            <td>1.0</td>
          </tr>
          <tr>
            <th scope="row">2</th>
            <td>EpiGe_logisticRegression</td>
            <td>1.0</td>
          </tr>
          <tr>
            <th scope="row">3</th>
            <td>EpiGe_hammingDistance</td>
            <td>1.0</td>
          </tr>
          <tr>
            <th scope="row">4</th>
            <td>EpiGe_report</td>
            <td>1.0</td>
          </tr>
        </tbody>
      </table>

    {% elif sample.status == 0 %}
    
    <h3>The file has not been analyzed.</h3>

    {% endif %}
{% endblock %}