{% load static %}
<html>
<body>
    <div class="container-logo">
            <img class="logos" src="{% static 'img/logos_report.png' %}" alt="logos">
    </div>
    <h2>EpiGe Medulloblastoma Subgroup Prediction Report</h2>
    <p>Sample ID: {{ sample.sample_identifier }}</p>
    <p>Uploaded file: {{ sample.filename}}</p>
    <p>Size file: {{ sample.filesize}}</p>
    <p>Instrument type: {{ calibration.instrument_type}}</p>
    <p>Email: {{ sample.email }}</p>
    <p>Date: {{ sample.created|date:"M d, Y, H:i:s" }}</p>

    <h3>EpiGe Medulloblastoma Subgroup Class Prediction (v1.1)</h3>
    <h3>Reference pattern data</h3>
    <div class="image-container">
        <img class="patterns" src="{% static 'img/tablaRef.png' %}" alt="patterns">
    </div>
    <h3>Results</h3>
    <p class="block-text">
    Sample pattern obtained by the logistic regression:
    </p>
    <div class="image-container">
        <img class="replicas" src="{{ classification.sample_image }}" alt="replicas">
    </div>
    <p class="block-text">
    Molecular subgroup obtained by the hamming distance:
    </p>
    <table>
        <thead>
        <tr>
            <th>Molecular subgroup</th>
            <th>Score WNT</th>
            <th>Score SHH</th>
            <th>Score non-WNT/non-SHH</th>
        </tr>
        </thead>
        <tbody>
        <tr>
            <td>{{ classification.subgroup }}</td>
            <td>{{ classification.score_wnt }}</td>
            <td>{{ classification.score_shh }}</td>
            <td>{{ classification.score_gg }}</td>
        </tr>
        </tbody>
    </table>
    <div class="image-container">
        <img class="radar" src="{{ classification.radar_plot }}" alt="replicas">
    </div>
    <h3>Subgroup Description</h3>
    <p><b>{{ classification.subgroup }}</b></p>
    <p class="block-text">
        {% if classification.subgroup == "WNT" %}
            The methylation class "medulloblastoma, WNT" is comprised of tumors diagnosed as Medulloblastoma, genetically defined, WNT-activated. Histologically most cases fall into the classic variant, sometimes into the large cell/anaplastic variant. Tumors are located in the cerebellum, typically in the vermis. Median age is 12 years (range 6 to 27). Monosomy of chromosome 6 (>80%) together with mutation of CTNNB1 is (almost) pathognomonic for this class.
        {% elif classification.subgroup == "SHH" %}
            The methylation class "medulloblastoma, subclass SHH‚Äù is comprised of tumors diagnosed as Medulloblastoma, genetically defined, SHH-activated. Histologically most cases fall into the desmoplastic variant or extensive nodularity variant, sometimes classic and occasionally large cell/anaplastic groups. Tumors are located in the cerebellum, usually laterally. Median age is is 2 years (range 0-6 years) for infants and 22 years (range 3 to 51) for children and adult. Upstream SHH pathway alterations (i.e. PTCH1 and SMO) or SUFU are relatively common, and in the youngest patients these may also have a germline origin (Gorlin's syndrome). Importantly, this methylation class also includes the majority of TP53-mutated SHH tumors (often Li-Fraumeni associated), which typically occur in children (~8-16 years) and often have large cell/anaplastic morphology, with dramatic copy number alterations (chromothripsis).
        {% else %}
            The methylation class "medulloblastoma, subclass non-WNT/non-SHH" is comprised of tumors with the diagnosis medulloblastoma, genetically defined, group 3 or group 4. Histologically most cases fall into the classical and large cell/anaplastic groups. Tumors are located in the cerebellum, typically in the vermis. non-WNT/non-SHH medulloblastomas are more common in males than females. MYC amplification, MYCN amplification, aneuploidy, isochromosome 17q and GFI1/1B activation by enhancer hijacking are recurrent features, but a fraction lack an obvious driving genetic change.
        {% endif %}
    </p>
    <h3>Disclaimer</h3>
    <p class="block-text">Classification using methylation profiling is a research tool under development, it is not verified and has not been clinically validated. Implementation of the results in a clinical setting is in the sole responsibility of the treating physician.</p>
    <p>Intended for non-commercial use only.</p>
    <h3>Cytosine Methylation Status Predictor</h3>
    {{ classification.logistic_table | safe }}
    <p class="block-text">
        Probability threshold: 0.456
    </p>
    <h3>Quality Control Summary</h3>
    <table>
        <thead>
        <tr>
            <th>CheckPoint</th>
            <th>Result</th>
            <th>Possible Solution</th>
        </tr>
        </thead>
        <tbody>
            <tr>
                <td>TXT complete</td>
                <td colspan="2">The txt file contains all the information.</td>
            </tr>
            <tr>
                <td>All CpG</td>
                <td colspan="2">All 6 CpG are included.</td>
            </tr>
            <tr>
                <td>Dye calibration</td>
                {% if calibration.ROX_valid and calibration.VIC_valid and calibration.FAM_valid %}
                    <td colspan="2">All dyes are calibrated.</td>
                {% else %}
                    <td>Some of the dyes have expired calibration.</td>
                    <td>See the technical report for more information.</td>
                {% endif %}
            </tr>
            <tr>
                <td>No template controls</td>
                {% if not calibration.amplification_test %}
                    <td>Amplification has been detected in some no template controls.</td>
                    <td>See the technical report for more information.</td>
                {% else %}
                    <td colspan="2">No amplification was detected in any no template control.</td>
                {% endif %}
            </tr>
            <tr>
                <td>SD<0.5</td>
                {% if calibration.standard_deviation_test %}
                    <td>Some pairs of replicates have a standard deviation greater than 0.5.</td>
                    <td>See the technical report for more information.</td>
                {% else %}
                    <td colspan="2">No pairs of replicates have a standard deviation greater than 0.5.</td>
                {% endif %}
            </tr>
        </tbody>
    </table>
    <div class="pagebreak"> </div>
    <h2>Technical report</h2>
    <h3>Calibration information</h3>
    <table>
        <thead>
        <tr>
            <th>ROX is expired</th>
            <th>ROX performed</th>
            <th>FAM is expired</th>
            <th>FAM performed</th>
            <th>VIC is expired</th>
            <th>VIC performed</th>
        </tr>
        </thead>
        <tbody>
            <tr>
                <td>{% if calibration.ROX_valid %}No{% else %}Yes{% endif %}</td>
                <td>{{ calibration.ROX_date }}</td>
                <td>{% if calibration.FAM_valid %}No{% else %}Yes{% endif %}</td>
                <td>{{ calibration.FAM_date }}</td>
                <td>{% if calibration.VIC_valid %}No{% else %}Yes{% endif %}</td>
                <td>{{ calibration.VIC_date }}</td>
            </tr>
        </tbody>
    </table>
    <h3>No template control amplification</h3>
    {{ calibration.amplification_table | safe }}
    {% if calibration.amplification_test %}
        No amplification detected.
    {% else %}
        Amplification detected.
    {% endif %}
    <h3>SD<0.5</h3>
    {{ calibration.standard_deviation_table | safe }}
    {% if calibration.standard_deviation_test %}
        Some pairs of replicates have a standard deviation greater than 0.5.
    {% else %}
        No pairs of replicates have a standard deviation greater than 0.5.
    {% endif %}
    <h3>Run information</h3>
    <table>
        <thead>
        <tr>
            <th>Task</th>
            <th>Version</th>
        </tr>
        </thead>
        <tbody>
            <tr>
                <td>EpiGe_qc</td>
                <td>1.0</td>
            </tr>
            <tr>
                <td>EpiGe_logisticRegression</td>
                <td>1.0</td>
            </tr>
            <tr>
                <td>EpiGe_randomForest</td>
                <td>1.0</td>
            </tr>
            <tr>
                <td>EpiGe_report</td>
                <td>1.0</td>
            </tr>
        </tbody>
    </table>
</body>
</html>