{% load static %}
<html>
<body>
    <div class="container-logo">
            <img class="logos" src="{% static 'img/logos_report.png' %}" alt="logos">
    </div>
    <h2>EpiGe Medulloblastoma Subgroup Prediction Report</h2>
    <p>Sample ID: {{ report.identifier }}</p>
    <p>Uploaded file: {{ report.filename}}</p>
    <p>Size file: {{ report.filesize}}</p>
    <p>Instrument type: {{ report.instrument_type}}</p>
    <p>Email: {{ report.email }}</p>
    <p>Date: {{ report.created|date:"M d, Y, H:i:s" }}</p>

    <h3>EpiGe Medulloblastoma Subgroup Class Prediction Results (v1.1)</h3>
    <p><b>Reference pattern data</b></p>
    <div class="image-container">
        <img class="patterns" src="{% static 'img/tablaRef.png' %}" alt="patterns">
    </div>
    <h3>Results</h3>
    {% if report.subgroup is not None %}
        <table>
            <thead>
            <tr>
                <th>Medulloblastoma predicted subgroup</th>
                <th>Score</th>
                <th>Interpretation</th>
            </tr>
            </thead>
            <tbody>
            <tr class="row{% cycle "1" "2" %}">
                <td>{{ report.subgroup }}</td>
                <td>{{ report.score }}</td>
                {% if report.score > 0.9 %}
                    <td><img class="points" src="{% static 'img/green_point.png' %}" alt="green_point"></td>
                {% else %}
                    <td><img class="points" src="{% static 'img/Red_point.png' %}" alt="red_point"></td>
                {% endif %}
            </tr>
            </tbody>
        </table>
        <p>Legend: <img class="points" src="{% static 'img/green_point.png' %}" alt="green_point">Match (score >= 0.9)<img class="points" src="{% static 'img/Red_point.png' %}" alt="green_point">Not match (score < 0.9)</p>
        <p><b>Patterns of the input data</b></p>
        <div class="image-container">
            <img class="replicas" src="{{ classification.replicas_image }}" alt="patterns">
        </div>
        <h3>Subgroup Description</h3>
        <p><b>{{ report.subgroup }}</b></p>
        <p class="block-text">
            {% if report.subgroup == "WNT" %}
                The methylation class "medulloblastoma, WNT" is comprised of tumors diagnosed as Medulloblastoma, genetically defined, WNT-activated. Histologically most cases fall into the classic variant, sometimes into the large cell/anaplastic variant. Tumors are located in the cerebellum, typically in the vermis. Median age is 12 years (range 6 to 27). Monosomy of chromosome 6 (>80%) together with mutation of CTNNB1 is (almost) pathognomonic for this class.
            {% elif report.subgroup == "SHH" %}
                The methylation class "medulloblastoma, subclass SHH‚Äù is comprised of tumors diagnosed as Medulloblastoma, genetically defined, SHH-activated. Histologically most cases fall into the desmoplastic variant or extensive nodularity variant, sometimes classic and occasionally large cell/anaplastic groups. Tumors are located in the cerebellum, usually laterally. Median age is is 2 years (range 0-6 years) for infants and 22 years (range 3 to 51) for children and adult. Upstream SHH pathway alterations (i.e. PTCH1 and SMO) or SUFU are relatively common, and in the youngest patients these may also have a germline origin (Gorlin's syndrome). Importantly, this methylation class also includes the majority of TP53-mutated SHH tumors (often Li-Fraumeni associated), which typically occur in children (~8-16 years) and often have large cell/anaplastic morphology, with dramatic copy number alterations (chromothripsis).
            {% else %}
                The methylation class "medulloblastoma, subclass non-WNT/non-SHH" is comprised of tumors with the diagnosis medulloblastoma, genetically defined, group 3 or group 4. Histologically most cases fall into the classical and large cell/anaplastic groups. Tumors are located in the cerebellum, typically in the vermis. non-WNT/non-SHH medulloblastomas are more common in males than females. MYC amplification, MYCN amplification, aneuploidy, isochromosome 17q and GFI1/1B activation by enhancer hijacking are recurrent features, but a fraction lack an obvious driving genetic change.
            {% endif %}
        </p>
    {% else %}
        This sample has not been classified.
    {% endif %}
    <h3>Disclaimer</h3>
    <p class="block-text">Classification using methylation profiling is a research tool under development, it is not verified and has not been clinically validated. Implementation of the results in a clinical setting is in the sole responsibility of the treating physician.<br> Intended for non-commercial use only.</p>
    <h3>Cytosine Methylation Status Predictor</h3>
    <img class="image-result" src="{{ classification.CMS_image }}"  alt="logo_sjd">
    {{ classification.CMS_table | safe }}
    <h3>Quality Control Summary</h3>
    <table>
        <thead>
        <tr>
            <th>CheckPoint</th>
            <th>Result</th>
            <th>Possible Solution</th>
        </tr>
        </thead>
        <tbody>
            <tr>
                <td>TXT complete</td>
                <td colspan="2">Yes</td>
            </tr>
            <tr>
                <td>All CpG</td>
                <td colspan="2">All 6 CpG are included</td>
            </tr>
            <tr>
                <td>Calibrators</td>
                <td colspan="2">Updated</td>
            </tr>
            <tr>
                <td>NTC</td>
                <td colspan="2">Not Amplified</td>
            </tr>
            <tr>
                <td>SD<0.25</td>
                <td colspan="2">5/6</td>
            </tr>
            <tr>
                <td>LDA</td>
                <td colspan="2"></td>
            </tr>
        </tbody>
    </table>
    <div class="pagebreak"> </div>
    <h2>Technical report</h2>
    <h3>Calibration information</h3>
    <table>
        <thead>
        <tr>
            <th>ROX is expired</th>
            <th>ROX performed</th>
            <th>FAM is expired</th>
            <th>FAM performed</th>
            <th>VIC is expired</th>
            <th>VIC performed</th>
        </tr>
        </thead>
        <tbody>
            <tr>
                <td>{% if report.ROX_valid %}No{% else %}Yes{% endif %}</td>
                <td>{{ report.ROX_date }}</td>
                <td>{% if report.FAM_valid %}No{% else %}Yes{% endif %}</td>
                <td>{{ report.FAM_date }}</td>
                <td>{% if report.VIC_valid %}No{% else %}Yes{% endif %}</td>
                <td>{{ report.VIC_date }}</td>
            </tr>
        </tbody>
    </table>
    <h3>NTC amplification</h3>
    {% if not report.amplification_test %}
        {{ calibration.amplification_table | safe }}
    {% else %}
        No amplification detected.
    {% endif %}
    <h3>Run information</h3>
    <table>
        <thead>
        <tr>
            <th>Task</th>
            <th>Version</th>
        </tr>
        </thead>
        <tbody>
            <tr>
                <td>EpiGe_qc</td>
                <td>1.0</td>
            </tr>
            <tr>
                <td>EpiGe_CMS</td>
                <td>1.0</td>
            </tr>
            <tr>
                <td>EpiGe_LDA</td>
                <td>1.0</td>
            </tr>
            <tr>
                <td>EpiGe_report</td>
                <td>1.0</td>
            </tr>
        </tbody>
    </table>
</body>
</html>